holonix/test/shell-setup.bats

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bats
+
+@test "shellHook is evaluated" {
+    [ -n $HOLOCHAIN_DEVSHELL ]
```

Comment:
Nice! That's a tidy and simple way to check that, I like it

Reply:
This always passes for me, regardless of whether `HOLOCHAIN_DEVSHELL` is set or not.

Reply:
how do you test that?

Reply:
I bats'd two test files like this:
```sh
$ cat test.bats 
#!/usr/bin/env bats

@test "variable" {
    [ -n $THIS_VARIABLE_DOES_DEFINITELY_NOT_EXIST ]
}

$ bats test.bats 
test.bats
 ✓ variable

1 test, 0 failures

$ cat test2.bats 
#!/usr/bin/env bats

@test "variable" {
    [ -n $SHELL ]
}

$ bats test2.bats 
test2.bats
 ✓ variable

1 test, 0 failures
```

Reply:
Ich habe auch das hier versucht:
```sh
$ cat test.bats 
#!/usr/bin/env bats

@test "variable" {
    unset THIS_VARIABLE_DOES_DEFINITELY_NOT_EXIST
    [ -n $THIS_VARIABLE_DOES_DEFINITELY_NOT_EXIST ]
}

$ bats test.bats 
test.bats
 ✓ variable

1 test, 0 failures

$ cat test2.bats 
#!/usr/bin/env bats

@test "variable" {
    SHELL="something"
    [ -n $SHELL ]
}

$ bats test2.bats 
test2.bats
 ✓ variable

1 test, 0 failures
```

Reply:
thanks, i figured it out. missing quotes

```suggestion
    [ -n "$HOLOCHAIN_DEVSHELL" ]
```

Reply:
Yes, that seems fine:
```sh
$ cat test.bats
#!/usr/bin/env bats

@test "variable" {
    [ -n "$THIS_VARIABLE_DOES_DEFINITELY_NOT_EXIST" ]
}

$ bats test.bats 
test.bats
 ✗ variable
   (in test file test.bats, line 4)
     `[ -n "$THIS_VARIABLE_DOES_DEFINITELY_NOT_EXIST" ]' failed

1 test, 1 failure

$ THIS_VARIABLE_DOES_DEFINITELY_NOT_EXIST="something" bats test.bats
test.bats
 ✓ variable

1 test, 0 failures
```

What doesn't work is:
```sh
$ THIS_VARIABLE_DOES_DEFINITELY_NOT_EXIST="something"
Desktop  $ echo $THIS_VARIABLE_DOES_DEFINITELY_NOT_EXIST 
something
Desktop  $ bats test.bats                                     
test.bats
 ✗ variable
   (in test file test.bats, line 4)
     `[ -n "$THIS_VARIABLE_DOES_DEFINITELY_NOT_EXIST" ]' failed

1 test, 1 failure
```
but I suppose that's because `bats` runs in a different shell where that env var isn't set. How does that behave in the above test?
```

Reply:
> but I suppose that's because `bats` runs in a different shell where that env var isn't set. How does that behave in the above test?

right, specifically because you set the variable without `export`ing it. i.e. `export THIS_VARIABLE_DOES_DEFINITELY_NOT_EXIST="something"` should cause that test to pass.